Assets/Scripts/MoveToClickPoint.cs

```diff
@@ -0,0 +1,56 @@
+// MoveToClickPoint.cs
+using UnityEngine;
+using UnityEngine.AI;
+    
+public class MoveToClickPoint : MonoBehaviour
+{
+    [SerializeField] private NavMeshAgent player;
+    [SerializeField] private Animator anim;
+    [SerializeField] private float distanceToTarget = 2f;
+    
+    private Transform target;
+    private Camera mainCam;
+        
+    void Awake() 
+    {
+        mainCam = Camera.main;
+        player = GetComponent<NavMeshAgent>();
+    }
+        
+    void Update()
+    {
+        anim.SetFloat("isWalking", player.velocity.magnitude/player.speed);
+
+        GetDestination();
+
+        if (target != null)
+        {
+            var between = target.position - transform.position;
+            
+            if(between.magnitude <= distanceToTarget)
+                return;
+            
+            player.SetDestination(target.position);
+        }
+    }
+
+    private void GetDestination()
+    {
+        if (!Input.GetMouseButtonDown(1)) 
```

Comment:
Keep in mind, I may not understand NavMesh movement fully, but would I think it would make more sense to check for the `mouse down` in update and only then do a GetDestination() -- or as I suggested above, something akin to `_curTarget = GetDestination()`

Reply:
I believe the logic is set up so it checks all 3 things every frame under one method, otherwise if we set up on mouse click, it would then go check in a separate method if the ray hit something, and if the ray hit something have to compare the tag if it is something the player should GoTo... when I was following the instructions it seemed to make sense to have it all in one, even if it seems weird to be checking if something isn't happening every frame as oppose to has something happened, just seems more streamlined. Should I re-write that?

Reply:
I am not 100% on NavMesh either, and really hesitate to change this entire logic. It might take a substantial amount of time for me to redo the logic

Reply:
It works - in the interest of time, resolving conversation.